diesel/src/expression_methods/bool_expression_methods.rs

```diff
@@ -43,6 +43,7 @@ pub trait BoolExpressionMethods: Expression + Sized {
         Self::SqlType: SqlType,
         ST: SqlType + TypedExpressionType,
         T: AsExpression<ST>,
+        <T::Expression as Expression>::SqlType: BoolOrNullableBool,
```

Comment:
It seems that this can be written as `ST: BoolOrNullableBool` given the other bounds